pandas/io/formats/latex.py

```diff
@@ -106,13 +112,60 @@ def pad_empties(x):
             raise AssertionError('column_format must be str or unicode, '
                                  'not {typ}'.format(typ=type(column_format)))
 
+        use_table_env = False
         if not self.longtable:
+            if self.caption is None and self.label is None:
+                # then write output only in a tabular environment
+                pass
+            else:
+                # then write output in a nested table/tabular environment
+                use_table_env = True
+
+                if self.caption is None:
+                    caption_ = ''
+                else:
+                    caption_ = '\n\\caption{{{}}}'.format(self.caption)
+
+                if self.label is None:
+                    label_ = ''
+                    warnings.warn('no LaTeX label has been provided; '
```

Comment:
Is there any strong reason for a warning here? I think it is fine to produce tables without labels and that most latex users will know they can't reference a table without a label.

Reply:
@tomneep I think you're right.  I will remove the warning.  thanks.